### PR TITLE
`CallResult` in trace

### LINKED ISSUE #1428: Add call result field to trace
Field with a serialised result of a call will make debugging easier and asserting trace more accurate

### PR DESCRIPTION
**Pull request by war-in**
Closes #1428 

## Introduced changes

- `CallResult` is available in `CallTrace` and `ProfilerCallTrace`

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/tests/e2e/trace_print.rs [hunk 1]
```diff
@@ -69,10 +72,13 @@ fn trace_info_print() {
                         Caller address: [..]
                         Call type: Call
                         Nested Calls: []
+                        Call Result: Success: []
                     )
                 ]
+                Call Result: Success: []
             )
         ]
+        Call Result: Success: []
```
Comment (drknzz) [inline crates/forge/tests/e2e/trace_print.rs:81]: Could we have an e2e test with Call Result: Failure? 👀

**Review by drknzz — COMMENTED** [review]
lgtm so far

File: crates/forge-runner/src/trace_data.rs [hunk 2]
```diff
@@ -33,6 +34,7 @@ pub struct ProfilerCallTrace {
     // These also include resources used by internal calls
     pub used_execution_resources: ProfilerExecutionResources,
     pub nested_calls: Vec<ProfilerCallTrace>,
+    pub result: CallResult,
```
Comment (piotmag769) [inline crates/forge-runner/src/trace_data.rs:37]: Just realised - we don't need it here, we don't use it in the profiler anyways

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs [hunk 3]
```diff
@@ -130,7 +132,22 @@ pub fn execute_call_entry_point(
...
+    runtime_state.cheatnet_state.trace_data.exit_nested_call(
+        resources,
+        CallResult::from_execution_result(
+            &result,
+            &match entry_point.call_type {
+                CallType::Call => AddressOrClassHash::ContractAddress(entry_point.storage_address),
+                CallType::Delegate => {
+                    AddressOrClassHash::ClassHash(entry_point.class_hash.unwrap())
+                }
+            },
```
Comment (piotmag769) [inline crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs:146]: Could be in a separate variable imo

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**